Library/Homebrew/livecheck/strategy/crates.rb

```diff
@@ -0,0 +1,111 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      # The {Crates} strategy identifies versions of a crates.io package by
+      # checking the information from the `versions` API endpoint.
+      #
+      # Crate URLs have the following format:
+      #   `https://static.crates.io/crates/example/example-1.2.3.crate`
+      #
+      # The default regex identifies versions like `1.2.3`/`v1.2.3` from the
+      # version `num` field. This is a common version format but a different
+      # regex can be provided in a `livecheck` block to override the default
+      # if a package uses a different format (e.g. `1.2.3d`, `1.2.3-4`, etc.).
+      #
+      # @api public
+      class Crates
```

Comment:
I think this should either be called `Crate` or `CratesIo`. Probably just `Crate` if we want to support third-party repositories in this class as mentioned in https://github.com/Homebrew/brew/pull/16620#discussion_r1483538134.

Reply:
Thinking about this some more, I agree that the generic `Crate` name would make sense for both crates.io and alternative registries. It may not be something that we support now but it would save us from dealing with a rename in the future, so I'll switch this back on the next push.